packages/pigeon/lib/dart_generator.dart

```diff
@@ -236,20 +236,21 @@ void _writeFlutterApi(
       );
       indent.writeln('$returnType ${func.name}($argSignature);');
     }
-    indent.write('static void setup(${api.name}$nullTag api) ');
+    indent.write(
+        'static void setup(${api.name}$nullTag api, {BinaryMessenger$nullTag binaryMessenger}) ');
```

Comment:
Is the addition of binaryMessenger related to this fix?

Reply:
It was required for the test for the fix.  I think we were relying on compilation / text-output unit tests for FlutterApi generation a lot and didn't have existing runtime tests for the Dart code generated from FlutterApi.  Less people use FlutterApi too so it probably went unnoticed because of that.